[ENH] Reorganize mask and scalar system into one "Image" system (#817)
yeatmanlab/pyAFQ: docs/source/usage/image.rst

The pyAFQ Image API
~~~~~~~~~~~~~~~~~~
In the process of tractometry, it is sometimes necessary to apply images to
the data. By default, pyAFQ can automatically generate these images from the
DWI data. However, pyAFQ also has a system for users to specify which image to
use, called the Image API. Specifying a image, in general, is not as simple as
just providing a path to a file, because each subject and each session will
have a different image. That is why we developed pyAFQ's Image API. 

Currently, there are three different images that pyAFQ uses for tractometry:

> #. The brain image. This is used to image the dwi data and throwout any signal

Reviewer: ```suggestion
#. The brain image. This is used to image the dwi data and throw out any signal
```

Reviewer: Why is this not called a "mask" here?

>    that is outside of the brain. It is typicall applied before fitting ODF

Reviewer: ```suggestion
   that is outside of the brain. It is typically applied before fitting ODF
```

   models. By default, it is calculated using :class:`AFQ.definitions.image.B0Image`.

#. The tractography seed image. This image determines where tractography is
   seeded. If it is floating point, the image is thresholded interally after

>    interpolation using the seed_threshold parameter. This is reccomended.

Reviewer: ```suggestion
   interpolation using the `seed_threshold` parameter. This is recommended.
```

   However, the seed image can aslo be a binary image. By default, the
   seed image is :class:`AFQ.definitions.image.ScalarImage` (best_scalar) where best_scalar is chosen by the API
   based on valid scalars (typically "dti_fa"). 

#. The tractography stop image. This image determines where tractography stops.
   If it is floating point, the image is thresholded interally after

>    interpolation using the stop_threshold parameter. This is reccomended.

Reviewer: ```suggestion
   interpolation using the stop_threshold parameter. This is recommended.
```

  image if you want a image to be based on a scalar that pyAFQ already
  calculates, like "dti_fa" or "dti_md".

- :class:`AFQ.definitions.image.ThresholdedScalarImage`: This image is similar to :class:`AFQ.definitions.image.ScalarImage`. It allows the user to
  also provide a lower and upper bound. The bounds threshold the scalar data.
  Note that for the tractography seed and stop images, thresholding
  should typically be done using the seed_threshold and stop_threshold
  parameters, not using an already thresholded image.

- :class:`AFQ.definitions.image.PFTImage`: A image for specifying the segmentations used in PFT. Should only
  be used as a stop image. It's three arguments are three other images, which
  specify the three segmentations: white matter, gray matter, and
  corticospinal fluid.

- :class:`AFQ.definitions.image.CombinedImage`: This class can be used to combine the other images. It takes
  a list of images and allows the user to specify whether they should be
  combined using a logical "and" or "or".

Here is an example of using the :class:`AFQ.definitions.image.RoiImage` and :class:`AFQ.definitions.image.LabelledImageFile` on the HCP

> data with the AFQ object::

Reviewer: ```suggestion
data with the GroupAFQ object::
```
